File: apps/mobile/modules/adaptfit-pose/ios/AdaptfitPoseModule.swift
import ExpoModulesCore

public class AdaptfitPoseModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AdaptfitPose") // keep the name identical to android so js stays shared.

    Function("isAvailable") {
      // ios still uses expo-camera preview. don't claim mediapipe until the view infers.
      return false
    }

    View(AdaptfitPoseView.self) {
      Events("onAngles")
      // accept the same props as android so js doesn't branch per platform.
      Prop("enabled") { (_: AdaptfitPoseView, _: Bool) in }
      Prop("facing") { (_: AdaptfitPoseView, _: String) in }
      Prop("leftLandmarks") { (_: AdaptfitPoseView, _: [Int]) in }
      Prop("rightLandmarks") { (_: AdaptfitPoseView, _: [Int]) in }
    }
  }
}

File: apps/mobile/modules/adaptfit-pose/android/src/main/java/expo/modules/adaptfitpose/AdaptfitPoseModule.kt
package expo.modules.adaptfitpose

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class AdaptfitPoseModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AdaptfitPose") // module name must match the js native-module string.

    Function("isAvailable") {
      // android custom builds ship mediapipe. expo go never loads this module.
      true
    }

    View(AdaptfitPoseView::class) {
      Events("onAngles") // derived angles + confidence only.

      Prop("enabled") { view: AdaptfitPoseView, enabled: Boolean? ->
        view.setTrackingEnabled(enabled ?: false)
      }

      Prop("facing") { view: AdaptfitPoseView, facing: String? ->
        view.setFacing(facing ?: "front")
      }

      Prop("leftLandmarks") { view: AdaptfitPoseView, landmarks: List<Int>? ->
        // joint indices for the angle triple, not a pose dump to js.
        view.setLeftLandmarks(landmarks)
      }

      Prop("rightLandmarks") { view: AdaptfitPoseView, landmarks: List<Int>? ->
        view.setRightLandmarks(landmarks)
      }
    }
  }
}
